OpenAPI/LearningHub.Nhs.OpenApi.Services/Services/ResourceService.cs

```diff
@@ -73,14 +79,52 @@ public async Task<ResourceReferenceWithResourceDetailsViewModel> GetResourceRefe
             }
         }
 
+        /// <summary>
+        /// the get by  resource reference id async.
+        /// </summary>
+        /// <param name="resourceId">the id.</param>
+        /// <param name="currentUserId">.</param>
+        /// <returns>the resource.</returns>
+        public async Task<ResourceMetadataViewModel> GetResourceById(int resourceId, int? currentUserId)
+        {
+            var resourceIdList = new List<int>() { resourceId };
+
+            var resource = (currentUserId == null ?
+                            await this.resourceRepository.GetResourcesFromIds(resourceIdList)
+                            : await this.resourceRepository.GetResourcesFromIds(resourceIdList, (int)currentUserId))
+                        .SingleOrDefault();
+
+            if (resource == null)
+            {
+                throw new Exception($"Resource with ID {resourceId} not found. Provided userId was {(currentUserId.HasValue ? currentUserId.Value.ToString() : "null")}");
+            }
+
+            return new ResourceMetadataViewModel(
+                resource.Id,
+                resource.CurrentResourceVersion?.Title ?? ResourceHelpers.NoResourceVersionText,
+                resource.CurrentResourceVersion?.Description ?? string.Empty,
+                resource.ResourceReference.Select(rr => new ResourceReferenceViewModel(
+                                                        rr.OriginalResourceReferenceId,
+                                                        rr.GetCatalogue(),
+                                                        this.learningHubService.GetResourceLaunchUrl(rr.OriginalResourceReferenceId)))
+                                                        .ToList(),
+                resource.GetResourceTypeNameOrEmpty(),
+                resource.CurrentResourceVersion?.ResourceVersionRatingSummary?.AverageRating ?? 0.0m,
+                ActivityStatusHelper.UserSummaryActvityStatus((ActivityStatusEnum?)resource?.ResourceActivity?.FirstOrDefault()?.ActivityStatusId) ?? string.Empty);
```

Comment:
I may be confused here. The state here appears to come from the first activity record for the user. The user may have had multiple attemps so the first activity record may not be the relavant one.
(appologies if I am confused & wrong)

Reply:
I think this is a good point. I will look back at the tables and the queries. I had been tempted to do singleOrDefault as I was expecting a single status per resource that got updated. Would be great actually if once I have had a look, if I could give you a call if you have familiarity with some of these tables, I had hoped to be able to visualize them or navigate them a bit more when I was looking at the task. If not firstOrDefault I expect it will be max but I will have a look.

Reply:
@Phil-NHS  added a call to discuss in detail with Dave.